Band Protocol Grant Application

buranapongb opened the pull request.

TtomaS7 commented:
Thank you for submitting your application to ink!ubator. We appreciate your interest and the effort you've put into your proposal.
As I review your application, we have a couple of queries that we'd like to clarify in order to better understand your project:

You've provided a project timeline estimation ranging from 2-4 months. The span of this timeframe is quite broad. Could you please specify more clearly, or perhaps provide a more narrowed down estimate?

The presented cost - does it account for the monitoring costs for any of the chains? If it does, could you please indicate for which chains and the duration this cost covers?

We're eager to learn more about your proposal, and your responses to these questions will greatly assist us in our assessment.
Thank you once again for your interest in ink!ubator!!

buranapongb commented:
> Thank you for submitting your application to ink!ubator. We appreciate your interest and the effort you've put into your proposal. As I review your application, we have a couple of queries that we'd like to clarify in order to better understand your project:
> 
> You've provided a project timeline estimation ranging from 2-4 months. The span of this timeframe is quite broad. Could you please specify more clearly, or perhaps provide a more narrowed down estimate?
> 
> The presented cost - does it account for the monitoring costs for any of the chains? If it does, could you please indicate for which chains and the duration this cost covers?
> 
> We're eager to learn more about your proposal, and your responses to these questions will greatly assist us in our assessment. Thank you once again for your interest in ink!ubator!!

buranapongb commented:
@TtomaS7

TtomaS7 commented:
Thanks for clarifications! 
I will assign curators for the review.

h4x3rotab commented on applications/band-grants.md:
Could you elaborate on the questions above?

buranapongb replied:
Provided the missing info on the updated doc.

h4x3rotab commented on applications/band-grants.md:
Do you expect to expand the oracle service to other ink! based blockchain like AlephZero?

buranapongb replied:
We are excited about the opportunity to expand our oracle service to additional ink!-based blockchains. However, it is important to clarify that the grant we are seeking will be exclusively allocated to support the development and maintenance of Astar WASM.

buranapongb replied:
Updated, see changes in line 56

h4x3rotab commented on applications/band-grants.md:
```diff
@@ -0,0 +1,164 @@
+- A brief description of the template building.
+    - BandChain is based on Cosmos SDK, designed to be compatible with all smart contract platforms and blockchain development frameworks. In a trustless and decentralized manner, BandChain does all the heavy lifting jobs of pulling data from external sources, aggregating, and packaging them into a format that is easy to use and cryptographically verified across multiple blockchains. Unlike gener ...
+- An indication of how your project relates to ink! ecosystem.
+    - Band Protocol has the potential to become a crucial infrastructure provider for the ink! Ecosystem. As projects develop on Astar, they will require reliable oracle services to deliver precise real-world data, and VRF to produce randomized results in their gaming and lottery mechanisms. By integrating with Astar, we can prepare and offer these essential services to Dapps building on the plat ...
```
Does Band plan to provide the full access to the standard dataset as well as the VRF features to ink contracts this time?

buranapongb replied:
We have the capability to offer up to 15 symbols from our standard dataset (https://data.bandprotocol.com/). The VRF will not be included in ink contracts at this time. Nevertheless, we are eager and willing to integrate our VRF into ink contracts in the near future.

h4x3rotab commented on applications/band-grants.md:
What's the relationship between the relayer and the signer? Could you elaborate on the signing process, i.e. how is the security model? Does it use threshold signature to create efficient transactions?

h4x3rotab reviewed (commented):
Excited to see infrastructure project like oracle in the inkubator program application. I've left a few questions inline. Please check. :D

h4x3rotab reviewed (commented):
Excited to see infrastructure project like oracle in the inkubator program application. I've left a few questions inline. Please check. :D

h4x3rotab commented:
Sorry for the late reply. I believe this is a strong application that benefits the ink ecosystem a lot. Just the last question: I see there's only one milestone defined in this proposal. So you will need to finish the full project and get the bounty for once. Just want to confirm if it's expected.

buranapongb commented:
> Sorry for the late reply. I believe this is a strong application that benefits the ink ecosystem a lot. Just the last question: I see there's only one milestone defined in this proposal. So you will need to finish the full project and get the bounty for once. Just want to confirm if it's expected.

Is it feasible to divide the grant into multiple phases? For instance, we could consider a two-phase approach: 20% upfront and the remaining amount upon task completion. Alternatively, we could establish smaller milestones, although we currently find that approach inconvenient.

h4x3rotab commented:
> > Sorry for the late reply. I believe this is a strong application that benefits the ink ecosystem a lot. Just the last question: I see there's only one milestone defined in this proposal. So you will need to finish the full project and get the bounty for once. Just want to confirm if it's expected.
> 
> Is it feasible to divide the grant into multiple phases? For instance, we could consider a two-phase approach: 20% upfront and the remaining amount upon task completion. Alternatively, we could establish smaller milestones, although we currently find that approach inconvenient.

I don't think upfront payment is possible, but it's good to split the big milestone to multiple smaller milestones. Just note that each milestone should have clear deliverables.

buranapongb commented:
> > > Sorry for the late reply. I believe this is a strong application that benefits the ink ecosystem a lot. Just the last question: I see there's only one milestone defined in this proposal. So you will need to finish the full project and get the bounty for once. Just want to confirm if it's expected.
> > 
> > 
> > Is it feasible to divide the grant into multiple phases? For instance, we could consider a two-phase approach: 20% upfront and the remaining amount upon task completion. Alternatively, we could establish smaller milestones, although we currently find that approach inconvenient.
> 
> I don't think upfront payment is possible, but it's good to split the big milestone to multiple smaller milestones. Just note that each milestone should have clear deliverables.

If we split the big milestone into smaller milestones, will the grant award be distributed evenly? Say we split into two tasks, will we receive 50% of the grant award we asked for right after we've completed each task?

h4x3rotab commented:
The amount of the number should be justified by the workload of each milestone. So you can split the big one to smaller ones, and propose the amount according to the workload.

buranapongb commented:
> The amount of the number should be justified by the workload of each milestone. So you can split the big one to smaller ones, and propose the amount according to the workload.

We've just split our task into two, please have a look and let us know if you have any comments!

h4x3rotab commented:
> > The amount of the number should be justified by the workload of each milestone. So you can split the big one to smaller ones, and propose the amount according to the workload.
> 
> We've just split our task into two, please have a look and let us know if you have any comments!

You might have forgot to push the latest update. Could you take a look? So far it only has one milestone.

buranapongb commented:
> > > The amount of the number should be justified by the workload of each milestone. So you can split the big one to smaller ones, and propose the amount according to the workload.
> > 
> > 
> > We've just split our task into two, please have a look and let us know if you have any comments!
> 
> You might have forgot to push the latest update. Could you take a look? So far it only has one milestone.

My apologies, it should be updated now.

HCastano commented on applications/band-grants.md:
```suggestion
```

HCastano commented on applications/band-grants.md:
Is there a GitHub profile for the other FTE you are asking for funding for?

HCastano commented on applications/band-grants.md:
```suggestion
```

HCastano commented on applications/band-grants.md:
Same thing here

HCastano reviewed (commented):
There's still a few things to clean up, but it's getting there

HCastano commented:
@buranapongb any updates to my questions 😄 ?

HCastano commented:
@buranapongb pinging again

buranapongb commented:
> @buranapongb pinging again

@HCastano I've updated the file, please let me know if there are any other comments.

HCastano commented on applications/band-grants.md:
I would like to see a Docker image provided here

HCastano commented on applications/band-grants.md:
The audit wouldn't be part of this milestone.

You should have unit tests and some sort of automated integration test suite

HCastano commented on applications/band-grants.md:
Why does the Astar EVM chain need to be monitored?

What tools are you planning to use here?

TtomaS7 commented:
Congratulations! We are thrilled to officially welcome you to the Ink!ubator program!! We can't wait to see the remarkable contributions you'll bring to the ecosystem.
In terms of the next steps, we are eagerly anticipating your first delivery, which is due in 2-3 months. When it’s ready, please create a PR here - https://github.com/use-inkubator/Grant-Milestone-Delivery

If you have any questions, concerns, or if you just want to brainstorm ideas, don't hesitate to reach out. We encourage you to utilize our Element channel for any inquiries - https://matrix.to/#/#sm-bounty-support:matrix.org

As you join the Ink!ubator family, please remember that you are not alone on this journey. We are here to support you every step of the way. We believe in your potential and we can't wait to see where this journey takes us all.

Congratulations once again and welcome aboard!
